openshift/sdn
# Bug 1984683: use new default leader election values to handle apiserver rollout on SNO

In case of SDN gets deployed on SNO, it can run into leader election timeouts during an api server rollout (as only one api server is available and it doesn't restart fast enough yet). 
Therefor the default values for leader election have been updated (https://github.com/openshift/library-go/pull/1104) to the [recommended](https://github.com/openshift/enhancements/blame/84e894ead7b188a1013556e0ba6973b8463995f1/CONVENTIONS.md#L183) values.

This PR updates the version of the openshift/library-go library to use these new values and being able to handle a 60 seconds downtime of the api server on SNO.

**How to reproduce / validate:**
1. Start a SNO cluster with this patch
2. Trigger a apiserver rollout (e.g. via `k patch kubeapiserver/cluster --type merge -p '{"spec":{"forceRedeploymentReason":"something"}}'`)
3. Check the logs from sdncontroller pod during the update
4. Validate the pod did not get restarted cause it lost the lease (e.g. check no previous logs are available / doesn't contain `leaderelection lost`

**Special notes for reviewers:**
* had to update the go version for the build root image: https://github.com/openshift/sdn/blob/d9796069710fffef11bf459ba85259e8b116166a/.ci-operator.yaml#L1-L4 as openshift/build-machinery required at least [go 1.15.2](https://github.com/openshift/build-machinery-go/blob/1bb7fd1518d37766cd8b8ea026c868fbb8e64349/make/lib/golang.mk#L20).
